diff --git a/src/BowlingGame.php b/src/BowlingGame.php
index 99575ea..1c44717 100644
--- a/src/BowlingGame.php
+++ b/src/BowlingGame.php
@@ -23,14 +23,14 @@ class BowlingGame
         $score = 0;
         $frameIndex = 0;
         for ($frame = 0; $frame < 10; $frame++) {
-            if ($this->rolls[$frameIndex] === 10) { // strike
-                $score += 10 + $this->rolls[$frameIndex + 1] + $this->rolls[$frameIndex + 2];
+            if ($this->isStrike($frameIndex)) {
+                $score += 10 + $this->strikeBonus($frameIndex);
                 $frameIndex++;
             } else if ($this->isSpare($frameIndex)) {
-                $score += 10 + $this->rolls[$frameIndex + 2];
+                $score += 10 + $this->spareBonus($frameIndex);
                 $frameIndex += 2;
             } else {
-                $score += $this->rolls[$frameIndex] + $this->rolls[$frameIndex + 1];
+                $score += $this->sumBallsInFrame($frameIndex);
                 $frameIndex += 2;
             }
         }
@@ -38,8 +38,28 @@ class BowlingGame
         return $score;
     }
 
-    protected function isSpare(int $frameIndex): bool
+    private function isStrike(int $frameIndex): bool
+    {
+        return $this->rolls[$frameIndex] === 10;
+    }
+
+    private function isSpare(int $frameIndex): bool
     {
         return $this->rolls[$frameIndex] + $this->rolls[$frameIndex + 1] === 10;
     }
+
+    private function strikeBonus(int $frameIndex): int
+    {
+        return $this->rolls[$frameIndex + 1] + $this->rolls[$frameIndex + 2];
+    }
+
+    private function spareBonus(int $frameIndex): int
+    {
+        return $this->rolls[$frameIndex + 2];
+    }
+
+    private function sumBallsInFrame(int $frameIndex): int
+    {
+        return $this->rolls[$frameIndex] + $this->rolls[$frameIndex + 1];
+    }
 }
diff --git a/tests/BowlingGameTest.php b/tests/BowlingGameTest.php
index c0c9bca..a7770b9 100644
--- a/tests/BowlingGameTest.php
+++ b/tests/BowlingGameTest.php
@@ -42,8 +42,7 @@ class BowlingGameTest extends TestCase
 
     public function testOneStrike()
     {
-        $this->game->roll(10); // strike
-        $this->game->roll(3);
+        $this->rollStrike();
         $this->game->roll(4);
         $this->rollMany(0, 16);
 
@@ -62,4 +61,10 @@ class BowlingGameTest extends TestCase
         $this->game->roll(5);
         $this->game->roll(5);
     }
+
+    private function rollStrike(): void
+    {
+        $this->game->roll(10);
+        $this->game->roll(3);
+    }
 }
