diff --git a/src/BowlingGame.php b/src/BowlingGame.php
new file mode 100644
index 0000000..2e891a9
--- /dev/null
+++ b/src/BowlingGame.php
@@ -0,0 +1,16 @@
+<?php
+
+namespace App;
+
+class BowlingGame
+{
+    public function roll(int $int)
+    {
+    }
+
+    public function score(): int
+    {
+        // "fake it till you make it" technique
+        return 0;
+    }
+}
diff --git a/tests/BowlingGameTest.php b/tests/BowlingGameTest.php
index 2f7af8e..ea7a867 100644
--- a/tests/BowlingGameTest.php
+++ b/tests/BowlingGameTest.php
@@ -2,9 +2,19 @@
 
 namespace App\Tests;
 
+use App\BowlingGame;
 use PHPUnit\Framework\TestCase;
 
 class BowlingGameTest extends TestCase
 {
-    
+    public function testGutterGame()
+    {
+        $game = new BowlingGame();
+
+        for ($i = 0; $i < 20; $i++) {
+            $game->roll(0);
+        }
+
+        self::assertEquals(0, $game->score());
+    }
 }
