diff --git a/src/NameInverter.php b/src/NameInverter.php
index f7cd091..6328e66 100644
--- a/src/NameInverter.php
+++ b/src/NameInverter.php
@@ -11,6 +11,6 @@ class NameInverter
         }
 
         // fake it till you make it
-        return $name;
+        return trim($name);
     }
 }
diff --git a/tests/NameInverterTest.php b/tests/NameInverterTest.php
index 44dbd69..585b560 100644
--- a/tests/NameInverterTest.php
+++ b/tests/NameInverterTest.php
@@ -31,6 +31,11 @@ class NameInverterTest extends TestCase
         $this->assertInvertedName('Diego', 'Diego');
     }
 
+    public function testInvert_givenSimpleNameWithSpaces_returnNameWithoutSpaces()
+    {
+        $this->assertInvertedName('  Diego  ', 'Diego');
+    }
+
     private function assertInvertedName(string $name, string $invertedName): void
     {
         self::assertEquals($invertedName, $this->nameInverter->invert($name));
