diff --git a/src/NameInverter.php b/src/NameInverter.php
index c68fd42..f7cd091 100644
--- a/src/NameInverter.php
+++ b/src/NameInverter.php
@@ -6,7 +6,11 @@ class NameInverter
 {
     public function invert(?string $name): string
     {
+        if (!$name) {
+            return '';
+        }
+
         // fake it till you make it
-        return '';
+        return $name;
     }
 }
diff --git a/tests/NameInverterTest.php b/tests/NameInverterTest.php
index 6512056..c07dca0 100644
--- a/tests/NameInverterTest.php
+++ b/tests/NameInverterTest.php
@@ -16,6 +16,14 @@ class NameInverterTest extends TestCase
 
         self::assertEquals('', $invertedName);
     }
-
     
+    public function testInvert_givenSimpleName_returnName()
+    {
+        $name = 'Diego';
+
+        $nameInverter = new NameInverter();
+        $invertedName = $nameInverter->invert($name);
+        
+        self::assertEquals('Diego', $invertedName);
+    }
 }
