diff --git a/src/NameInverter.php b/src/NameInverter.php
index 8bb6136..f801055 100644
--- a/src/NameInverter.php
+++ b/src/NameInverter.php
@@ -10,7 +10,7 @@ class NameInverter
             return '';
         }
 
-        $nameParts = explode(' ', trim($name));
+        $nameParts = preg_split('/\s+/', trim($name));
         if (count($nameParts) === 1) {
             return $nameParts[0];
         }
diff --git a/tests/NameInverterTest.php b/tests/NameInverterTest.php
index 0490e6b..60d65c7 100644
--- a/tests/NameInverterTest.php
+++ b/tests/NameInverterTest.php
@@ -41,6 +41,11 @@ class NameInverterTest extends TestCase
         $this->assertInvertedName('Diego Aguiar', 'Aguiar, Diego');
     }
     
+    public function testInvert_givenFirstLastWithSpaces_returnLastFirstWithoutSpaces()
+    {
+        $this->assertInvertedName('  Diego  Aguiar  ', 'Aguiar, Diego');
+    }
+    
     private function assertInvertedName(string $name, string $invertedName): void
     {
         self::assertEquals($invertedName, $this->nameInverter->invert($name));
