diff --git a/src/NameInverter.php b/src/NameInverter.php
new file mode 100644
index 0000000..c68fd42
--- /dev/null
+++ b/src/NameInverter.php
@@ -0,0 +1,12 @@
+<?php
+
+namespace App;
+
+class NameInverter
+{
+    public function invert(?string $name): string
+    {
+        // fake it till you make it
+        return '';
+    }
+}
diff --git a/tests/NameInverterTest.php b/tests/NameInverterTest.php
new file mode 100644
index 0000000..f74cf8b
--- /dev/null
+++ b/tests/NameInverterTest.php
@@ -0,0 +1,22 @@
+<?php
+
+namespace App\Tests;
+
+use App\NameInverter;
+use PHPUnit\Framework\TestCase;
+
+class NameInverterTest extends TestCase
+{
+    public function testInvert_givenNull_returnEmptyString()
+    {
+        // Arrange
+        $name = null;
+
+        // Act
+        $nameInverter = new NameInverter();
+        $invertedName = $nameInverter->invert($name);
+
+        // Assert
+        self::assertEquals('', $invertedName);
+    }
+}
