diff --git a/src/NameInverter.php b/src/NameInverter.php
index 6328e66..8bb6136 100644
--- a/src/NameInverter.php
+++ b/src/NameInverter.php
@@ -10,7 +10,11 @@ class NameInverter
             return '';
         }
 
-        // fake it till you make it
-        return trim($name);
+        $nameParts = explode(' ', trim($name));
+        if (count($nameParts) === 1) {
+            return $nameParts[0];
+        }
+
+        return sprintf('%s, %s', $nameParts[1], $nameParts[0]);
     }
 }
diff --git a/tests/NameInverterTest.php b/tests/NameInverterTest.php
index 585b560..0490e6b 100644
--- a/tests/NameInverterTest.php
+++ b/tests/NameInverterTest.php
@@ -36,6 +36,11 @@ class NameInverterTest extends TestCase
         $this->assertInvertedName('  Diego  ', 'Diego');
     }
 
+    public function testInvert_givenFirstLast_returnLastFirst()
+    {
+        $this->assertInvertedName('Diego Aguiar', 'Aguiar, Diego');
+    }
+    
     private function assertInvertedName(string $name, string $invertedName): void
     {
         self::assertEquals($invertedName, $this->nameInverter->invert($name));
