diff --git a/src/Entity/GameResult.php b/src/Entity/GameResult.php
index 8270314..4fdd096 100644
--- a/src/Entity/GameResult.php
+++ b/src/Entity/GameResult.php
@@ -59,4 +59,12 @@ class GameResult
     {
         return $this->createdAt;
     }
+
+    /**
+     * @internal only for testing purposes
+     */
+    public function setCreatedAt(\DateTimeImmutable $createdAt)
+    {
+        $this->createdAt = $createdAt;
+    }
 }
diff --git a/src/Repository/PlayerRepository.php b/src/Repository/PlayerRepository.php
index fcca292..e324f57 100644
--- a/src/Repository/PlayerRepository.php
+++ b/src/Repository/PlayerRepository.php
@@ -22,6 +22,9 @@ class PlayerRepository extends ServiceEntityRepository
     public function getScoreboard(): array
     {
         return $this->createQueryBuilder('p')
+            ->join('p.gameResults', 'gr')
+            ->andWhere('gr.createdAt >= :dateLimit')
+            ->setParameter('dateLimit', new \DateTimeImmutable('-15 days'))
             ->setMaxResults(5)
             ->getQuery()
             ->execute();
diff --git a/tests/Repository/PlayerRepositoryTest.php b/tests/Repository/PlayerRepositoryTest.php
index e334ae2..9f62732 100644
--- a/tests/Repository/PlayerRepositoryTest.php
+++ b/tests/Repository/PlayerRepositoryTest.php
@@ -63,6 +63,20 @@ class PlayerRepositoryTest extends WebTestCase
         self::assertCount(5, $scoreboard);
     }
 
+    public function testGetScoreBoard_ignoresRecordsOlderThan15Days()
+    {
+        $this->createPlayerWithVictory('player1');
+        $p2 = $this->createPlayer('p2');
+        $gameResult = $this->createGameResult($p2, true);
+        $gameResult->setCreatedAt(new \DateTimeImmutable('-16 days'));
+
+        $this->getEntityManager()->flush();
+
+        $scoreboard = $this->repository->getScoreboard();
+        
+        self::assertCount(1, $scoreboard);
+    }
+    
     private function getEntityManager(): EntityManagerInterface
     {
         return self::getContainer()->get(EntityManagerInterface::class);
