diff --git a/tests/Repository/PlayerRepositoryTest.php b/tests/Repository/PlayerRepositoryTest.php
new file mode 100644
index 0000000..45c7c8c
--- /dev/null
+++ b/tests/Repository/PlayerRepositoryTest.php
@@ -0,0 +1,20 @@
+<?php
+
+namespace App\Tests\Repository;
+
+use App\Repository\GameResultRepository;
+use App\Repository\PlayerRepository;
+use PHPUnit\Framework\TestCase;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+
+class PlayerRepositoryTest extends WebTestCase
+{
+    public function testSetup()
+    {
+        self::bootKernel();
+
+        $repository = self::getContainer()->get(PlayerRepository::class);
+
+        self::assertInstanceOf(PlayerRepository::class, $repository);
+    }
+}
