diff --git a/src/Repository/PlayerRepository.php b/src/Repository/PlayerRepository.php
index 4beda01..a2296be 100644
--- a/src/Repository/PlayerRepository.php
+++ b/src/Repository/PlayerRepository.php
@@ -18,4 +18,9 @@ class PlayerRepository extends ServiceEntityRepository
     {
         parent::__construct($registry, Player::class);
     }
+
+    public function getScoreboard(): array
+    {
+        return [];
+    }
 }
diff --git a/tests/Repository/PlayerRepositoryTest.php b/tests/Repository/PlayerRepositoryTest.php
index 45c7c8c..cf1f802 100644
--- a/tests/Repository/PlayerRepositoryTest.php
+++ b/tests/Repository/PlayerRepositoryTest.php
@@ -2,19 +2,28 @@
 
 namespace App\Tests\Repository;
 
-use App\Repository\GameResultRepository;
 use App\Repository\PlayerRepository;
-use PHPUnit\Framework\TestCase;
 use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
 
 class PlayerRepositoryTest extends WebTestCase
 {
-    public function testSetup()
+    // ZOMBIE (ZERO, ONE, MANY, BOUNDARIES, INTERFACES, EXCEPTIONS)
+    // scoreboard with zero players
+    // scoreboard with one player
+    // scoreboard with many players
+    // scoreboard shows no more than 5 players
+    // scoreboard ordered by highest score
+    // scoreboard is date range constrained to 15 days
+
+    public function testGetScoreboard_withZeroPlayers()
     {
         self::bootKernel();
 
+        /** @var PlayerRepository $repository */
         $repository = self::getContainer()->get(PlayerRepository::class);
 
-        self::assertInstanceOf(PlayerRepository::class, $repository);
+        $scoreboard = $repository->getScoreboard();
+
+        self::assertEmpty($scoreboard);
     }
 }
