diff --git a/src/WordWrapper.php b/src/WordWrapper.php
index 32edd1a..3e77f26 100644
--- a/src/WordWrapper.php
+++ b/src/WordWrapper.php
@@ -14,7 +14,7 @@ class WordWrapper
             return $s;
         }
 
-        $breakPoint = strrpos($s, ' ');
+        $breakPoint = strrpos(substr($s, 0, $width), ' ');
         if ($breakPoint === false) {
             $breakPoint = $width;
         }
diff --git a/tests/WordWrapperTest.php b/tests/WordWrapperTest.php
index ff9ead9..3e73d1a 100644
--- a/tests/WordWrapperTest.php
+++ b/tests/WordWrapperTest.php
@@ -25,7 +25,8 @@ class WordWrapperTest extends TestCase
             'break word' => ['xx', 1, "x\nx"],
             'one word multiple break lines' => ['xxx', 1, "x\nx\nx"],
             'space after break line' => ['x x', 1, "x\nx"],
-            'width includes a space' => ['x xx', 3, "x\nxx"]
+            'width includes a space' => ['x xx', 3, "x\nxx"],
+            'width includes a space following by another word' => ['x xx x', 3, "x\nxx\nx"],
         ];
     }
 }
