diff --git a/src/WordWrapper.php b/src/WordWrapper.php
index 3af112e..4515b24 100644
--- a/src/WordWrapper.php
+++ b/src/WordWrapper.php
@@ -14,6 +14,6 @@ class WordWrapper
             return $s;
         }
 
-        return substr($s, 0, $width) . "\n" . $this->wrap(substr($s, $width), $width);
+        return substr($s, 0, $width) . "\n" . $this->wrap(trim(substr($s, $width)), $width);
     }
 }
diff --git a/tests/WordWrapperTest.php b/tests/WordWrapperTest.php
index a243d7a..6bf4ff5 100644
--- a/tests/WordWrapperTest.php
+++ b/tests/WordWrapperTest.php
@@ -23,7 +23,8 @@ class WordWrapperTest extends TestCase
             'empty case' => ['', 1, ''],
             'one word' => ['x', 1, 'x'],
             'break word' => ['xx', 1, "x\nx"],
-            'one word multiple break lines' => ['xxx', 1, "x\nx\nx"]
+            'one word multiple break lines' => ['xxx', 1, "x\nx\nx"],
+            'space after break line' => ['x x', 1, "x\nx"]
         ];
     }
 }
