diff --git a/src/WordWrapper.php b/src/WordWrapper.php
index 44109ac..3af112e 100644
--- a/src/WordWrapper.php
+++ b/src/WordWrapper.php
@@ -14,6 +14,6 @@ class WordWrapper
             return $s;
         }
 
-        return substr($s, 0, $width) . "\n" . substr($s, $width);
+        return substr($s, 0, $width) . "\n" . $this->wrap(substr($s, $width), $width);
     }
 }
diff --git a/tests/WordWrapperTest.php b/tests/WordWrapperTest.php
index f8f1ea2..a243d7a 100644
--- a/tests/WordWrapperTest.php
+++ b/tests/WordWrapperTest.php
@@ -23,6 +23,7 @@ class WordWrapperTest extends TestCase
             'empty case' => ['', 1, ''],
             'one word' => ['x', 1, 'x'],
             'break word' => ['xx', 1, "x\nx"],
+            'one word multiple break lines' => ['xxx', 1, "x\nx\nx"]
         ];
     }
 }
