diff --git a/src/WordWrapper.php b/src/WordWrapper.php
index 1d99125..6d2a9e9 100644
--- a/src/WordWrapper.php
+++ b/src/WordWrapper.php
@@ -6,6 +6,10 @@ class WordWrapper
 {
     public function wrap(string $s, int $width): string
     {
-        return str_replace(' ', "\n", $s);
+        if (strlen($s) > $width) {
+            return str_replace(' ', "\n", $s);
+        }
+        
+        return $s;
     }
 }
diff --git a/tests/WordWrapperTest.php b/tests/WordWrapperTest.php
index 34418cf..5971b93 100644
--- a/tests/WordWrapperTest.php
+++ b/tests/WordWrapperTest.php
@@ -14,4 +14,12 @@ class WordWrapperTest extends TestCase
 
         self::assertEquals("word\nword", $result);
     }
+
+    public function testShouldNotWrap()
+    {
+        $wordWrapper = new WordWrapper();
+        $result = $wordWrapper->wrap('a word', 6);
+
+        self::assertEquals("a word", $result);
+    }
 }
