diff --git a/tests/WordWrapperTest.php b/tests/WordWrapperTest.php
index da27e1d..ba120ba 100644
--- a/tests/WordWrapperTest.php
+++ b/tests/WordWrapperTest.php
@@ -10,24 +10,8 @@ class WordWrapperTest extends TestCase
     public function testShouldWrap()
     {
         $wordWrapper = new WordWrapper();
-        $result = $wordWrapper->wrap('word word', 4);
+        $result = $wordWrapper->wrap('', 1);
 
-        self::assertEquals("word\nword", $result);
-    }
-
-    public function testShouldNotWrap()
-    {
-        $wordWrapper = new WordWrapper();
-        $result = $wordWrapper->wrap('a word', 6);
-
-        self::assertEquals("a word", $result);
-    }
-
-    public function testBreakLineAcrossWord()
-    {
-        $wordWrapper = new WordWrapper();
-        $result = $wordWrapper->wrap('this is a test', 4);
-
-        self::assertEquals("this\nis a\ntest", $result);
+        self::assertEquals('', $result);
     }
 }
