diff --git a/src/WordWrapper.php b/src/WordWrapper.php
index 6d2a9e9..44109ac 100644
--- a/src/WordWrapper.php
+++ b/src/WordWrapper.php
@@ -6,10 +6,14 @@ class WordWrapper
 {
     public function wrap(string $s, int $width): string
     {
-        if (strlen($s) > $width) {
-            return str_replace(' ', "\n", $s);
+        if (!$s) {
+            return '';
         }
         
-        return $s;
+        if (strlen($s) <= $width) {
+            return $s;
+        }
+
+        return substr($s, 0, $width) . "\n" . substr($s, $width);
     }
 }
diff --git a/tests/WordWrapperTest.php b/tests/WordWrapperTest.php
index 09c97e2..f8f1ea2 100644
--- a/tests/WordWrapperTest.php
+++ b/tests/WordWrapperTest.php
@@ -22,6 +22,7 @@ class WordWrapperTest extends TestCase
         return [
             'empty case' => ['', 1, ''],
             'one word' => ['x', 1, 'x'],
+            'break word' => ['xx', 1, "x\nx"],
         ];
     }
 }
